### DISCUSSION TITLE
How to use Fairy Chess with Stockfish in python for variant Antichess?

### DISCUSSION THREAD

**Discussion by JernejHenigman** [fairy-stockfish/Fairy-Stockfish#359] [Q&A] [answered]
Excuse me if this is not the best place to ask question like mine, and feel free to close the issue if not asked in the right place, but please if possible redirect me to the right address.

My question is how to utilize python Stockfish to support playing the game variant Antichess in python? Fairy chess is supposed to support AntiChess variant, but I am not sure how to utilize it for Stockfish?

With kinds regards
Jernej

**Comment by ianfab** [comment, marked as answer]
Hi, I converted the issue to a discussion as issues are reserved for feature requests and bug reports. You can ask questions either here in the discussions or on the discord server linked on the project page.

Now back to the question. If by "playing" you refer only to the support of the rules, then you can set up antichess positions using pyffish and e.g. let it return the available legal moves. Perhaps the easiest way to get into how to use it is to have a look at the unit tests https://github.com/ianfab/Fairy-Stockfish/blob/master/test.py.

Actually running the engine, i.e., letting it evaluate the position and determine the best move, is only possible with the native engine. If you want to build a simple script or app to play against the engine, then the usual way of doing that would be to maintain the board state using the python library (pyffish) and then communicate with a process of the native engine to request it to find a good move.

**Reply by JernejHenigman** [reply]
And how do I maintain board state of antichess and then communicate with stockfish native engine to suggest only possible moves for game of antichess? I am using python-chess, stockfish and pyffish, all running from python?

Regrards
Jernej

**Reply by ianfab** [reply]
If you are using python-chess, pyffish basically is redundant. Pyffish provides only a subset of features of python-chess, but it does it for many more variants. Therefore, if you only want to apply it to variants supported by python-chess, like antichess/giveaway, there is no need to use pyffish.

Python-chess has extensive documentation, also on how to communicate with an engine https://python-chess.readthedocs.io/en/latest/engine.html, so you should find what you need there. If you instead want to use pyffish, you can e.g. have a look at https://github.com/ianfab/fairyfishtest/blob/master/fairyfishtest.py as an example.

**Reply by JernejHenigman** [reply]
Thank you very much ianfab, I got what I wanted by running this code :)

engine = chess.engine.SimpleEngine.popen_uci("path/to/stockfish-windows-x86_64")
result = engine.play(board.board, chess.engine.Limit(time=0.001))
move = str(result.move)


Last question,
is it possible to set ELO for that engine programatically? 

Regards
Jernej

**Reply by ianfab** [reply]
See https://github.com/ianfab/Fairy-Stockfish#uci_limitstrength. How to set UCI options should be covered by python-chess docs.

**Reply by JernejHenigman** [reply]
Thank you that will do. I do not mind if the discussion is closed now 

Best
Jernej
